Documents/Command/AWS-RunAnsible/AWS-RunAnsible.yaml

```diff
@@ -0,0 +1,189 @@
+      bucket
+    default: ssm-playbook.yml
+  extravars:
+    type: String
+    description: '(Optional) Additional variables to pass to Ansible at runtime. Enter
+      a space separated list of key/value pairs. For example: color=red flavor=lime'
+    default: SSM=True
+    displayType: textarea
+    allowedPattern: ^$|^\w+\=[^\s|:();&]+(\s\w+\=[^\s|:();&]+)*$
+  WorkingDirectory:
+    type: String
+    description: ' (Optional) Current working directory to use for execution; if not
+      specified, use random temp directory.'
+    allowedPattern: ^\s*$|^[^']+$
+    default: ''
+  check:
+    type: String
+    description: ' (Optional) Use the check parameter to perform a dry run of the
+      Ansible execution.'
+    allowedValues:
+    - 'True'
+    - 'False'
+    default: 'False'
+  verbose:
+    type: String
+    description: ' (Optional) Set the level of verbosity for logging the execution
+      of the playbook. -v for low verbosity, -vvvv for debug level.'
+    allowedValues:
+    - -v
+    - -vv
+    - -vvv
+    - -vvvv
+    default: -v
+mainSteps:
+- action: aws:runShellScript
+  name: runShellScript
+  inputs:
+    runCommand:
+    - '#!/bin/bash'
+    - 'echo "Installing and or updating required tools: Ansible, wget unzip ...."
+      >&2'
+    - 'if cat /etc/issue|grep -i Amazon ; then '
+    - '   sudo pip install ansible --upgrade'
+    - '   sudo pip install awscli --upgrade'
+    - 'elif cat /etc/issue|grep -i Ubuntu ; then '
+    - '   sudo apt-get update'
+    - '   sudo DEBIAN_FRONTEND=noninteractive apt-get install python-pip -y'
+    - '   sudo pip install ansible --upgrade'
+    - '   sudo DEBIAN_FRONTEND=noninteractive apt-get install unzip -y'
+    - '   sudo DEBIAN_FRONTEND=noninteractive apt-get install awscli -y'
+    - 'elif cat /etc/issue|grep -i ''\\S'' ; then '
+    - '   sudo yum install -y https://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm'
+    - '   sudo yum -y update'
+    - '   sudo yum -y install python-pip'
+    - '   sudo pip install ansible --upgrade'
+    - '   sudo yum -y install wget'
+    - '   sudo yum -y install unzip'
+    - '   sudo pip install awscli --upgrade'
+    - else
+    - '   echo "There was a problem installin or updating the required tools for the
+      document. Plesae review the logs and correct the problem" >&2'
+    - '   exit 1'
+    - fi
+    - 'if [ -z "{{WorkingDirectory}}" ] ; then '
+    - '  cd `mktemp -d`'
+    - else
+    - '  mkdir -p {{WorkingDirectory}}; cd {{WorkingDirectory}}'
+    - fi
+    - echo "Running Ansible in `pwd`"
+    - '#if [ -z "${PlaybookFile}" ]'
+    - '#then'
+    - '#   PlaybookFile="ssm-playbook.yml"'
+    - '#else'
+    - PlaybookFile="{{PlaybookFile}}"
+    - '#fi'
+    - if [ -z "{{playbook}}" ] ; then
+    - '   PlaybookURL="{{PlaybookURL}}"'
+    - '   filename=${PlaybookURL##*/}'
+    - '   extension=${filename##*.}'
+    - '   if [[ "${PlaybookURL}" == http* ]]; then'
+    - '   if [[ "{{PlaybookURL}}" == */ ]]; then'
```

Comment:
Can we catch this invalid value in the allowedPattern of the PlaybookURL parameter somehow?